nameko_amqp_retry/backoff.py

```diff
@@ -133,10 +134,14 @@ def republish(self, backoff_exc, message, target_queue):
         # force redeclaration; the publisher will skip declaration if
         # the entity has previously been declared by the same connection
         # (see https://github.com/celery/kombu/pull/884)
+        # kombu >= 4.3.0 will raise NotBoundError here
         conn = Connection(amqp_uri)
-        maybe_declare(
-            queue, conn.channel(), retry=True, **DEFAULT_RETRY_POLICY
-        )
+        try:
+            maybe_declare(
+                queue, conn.channel(), retry=True, **DEFAULT_RETRY_POLICY
+            )
+        except NotBoundError:
+            pass
```

Comment:
@mattbennett - do you think it is worth handling multiple kombu versions like this? Or shall we just remove the line and force kombu to be > 4.3.0?

Reply:
we could log a warning or something here, so the exception wasn't entirely silent...

Reply:
I think Kombu 4.3.0 is too new to require. I prefer an explicit check on `kombu.__version__`. If we combine that with a link to the PR it should be very clear why the feature sniffing is there, and obvious to remove again once we stop supporting older versions.